core/jvm/src/test/scala/scalaz/zio/FiberSpec.scala

```diff
@@ -10,8 +10,9 @@ class FiberSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRun
   def e1 = unsafeRun(
     for {
       ref   <- Ref.make(false)
-      fiber <- IO.unit.bracket(_ => ref.set(true))(_ => IO.never).fork
-      _     <- fiber.toManaged.use(_ => IO.unit)
+      latch <- Promise.make[Nothing, Unit]
+      fiber <- IO.unit.bracket(_ => ref.set(true) *> latch.succeed(()))(_ => IO.never).fork
+      _     <- fiber.toManaged.use(_ => latch.await)
```

Comment:
One more thing if you want (only if you want!): to make this test better, after the `value <- ref.get`, we could `await` the fiber. if it has been successfully interrupted, then it will return right away; but if it has not been interrupted, then the `await` will time out because the fiber will still be in `IO.never`.

Reply:
I added it and the test now returns timeout :(

Reply:
Can I see code?

Reply:
Anyway, this is a huge improvement and will make the test pass reliably, I think. So 👍

Reply:
```scala
for {
      ref   <- Ref.make(false)
      latch <- Promise.make[Nothing, Unit]
      fiber  <- IO.unit.bracket(_ => ref.set(true) *> latch.succeed(()))(_ => IO.never).fork
      _        <- fiber.toManaged.use(_ => latch.await)
      _        <- fiber.await
      value <- ref.get
    } yield value must beTrue
```

Reply:
I am amazed this times out the test... because if you look at `toManaged`, it calls `interrupt` on `Fiber`, which cannot return until `Fiber` is done. So `await` should return immediately after the `use`. But we can investigate that some other time. It's enough for now to get the test reliably passing.

Reply:
It sounds like the finalizer is not interruptible and we have interrupt the fiber there..

Reply:
I was confused (it's wrong what I said -_- ^^

Reply:
How about this one:

```scala
for {
      ref   <- Ref.make(false)
      latch <- Promise.make[Nothing, Unit]
      fiber  <- IO.unit.bracket(_ => ref.set(true))(_ => latch.succeed(()) *> IO.never).fork
      _        <- fiber.toManaged.use(_ => latch.await)
      _        <- fiber.await
      value <- ref.get
    } yield value must beTrue
```

Reply:
I see now (I think 😆 ). The race is when the fiber starts execution!

So this should work:

```scala
How about this one:

for {
      ref   <- Ref.make(false)
      latch <- Promise.make[Nothing, Unit]
      fiber <- (latch.succeed(()) *> IO.unit).bracket_(ref.set(true))(IO.never).fork
      _     <- latch.await
      _     <- fiber.toManaged.use(_ => IO.unit)
      _     <- fiber.await
      value <- ref.get
    } yield value must beTrue
```

Reply:
Yes!! It works now, I will push that
Thank you! :)